data/core/config.lua

```diff
@@ -21,46 +76,188 @@ config.ignore_files = {
   "%.suo$",         "%.pdb$",       "%.idb$",        "%.class$", "%.psd$", "%.db$",
   "^desktop%.ini$", "^%.DS_Store$", "^%.directory$",
 }
+
+---Lua pattern used to find symbols when advanced syntax highlighting
+---is not available.
+---
+---The default pattern matches an alphanumeric character followed by
+---other alphabetic characters.
+---@type string
 config.symbol_pattern = "[%a_][%w_]*"
+
+---A list of characters that delimits a word.
+---
+---The default is `" \t\n/\\()\"':,.;<>~!@#$%^&*|+=[]{}`?-"`
+---@type string
 config.non_word_chars = " \t\n/\\()\"':,.;<>~!@#$%^&*|+=[]{}`?-"
+
+---The timeout, in seconds, before several consecutive actions
+---are merged as a single undo step.
+---
+---The default is 0.3 seconds.
+---@type number
 config.undo_merge_timeout = 0.3
+
+---The maximum number of undo steps per-document.
+---
+---The default is 10000.
+---@type number
 config.max_undos = 10000
+
+---The maximum number of tabs shown at a time.
+---
+---The default is 8.
+---@type number
 config.max_tabs = 8
+
+---Shows/hides the tab bar when there is only one tab open.
+---
+---The tab bar is always shown by default.
+---@type boolean
 config.always_show_tabs = true
+
+---@alias config.highlightlinetype
+---| true # Always highlight the current line.
+---| false # Never highlight the current line.
+---| "no_selection" Highlight the current line if no text is selected.
+
 -- Possible values: false, true, "no_selection"
+---Highlights the current line.
+---
+---The default is true.
+---@type config.highlightlinetype
 config.highlight_current_line = true
+
+---The spacing between each line of text.
+---
+---The default is 120% of the height of the text (1.2).
+---@type number
 config.line_height = 1.2
+
+---The number of spaces each level of indentation represents.
+---
+---The default is 2.
+---@type number
 config.indent_size = 2
+
+---The type of indentation.
+---
+---The default is "soft" (spaces).
+---@type "soft" | "hard"
 config.tab_type = "soft"
+
+---Do not remove whitespaces when advancing to the next line.
+---
+---Defaults to false.
+---@type boolean
 config.keep_newline_whitespace = false
+
+---Maximum number of characters per-line.
+---
+---Defaults to 80.
+---@type number
 config.line_limit = 80
+
+---Maximum number of project files to keep track of.
+---If the number of files in the project exceeds this number,
+---Lite XL will not be able to keep track of them.
+---It will not show in the file picker and any changes will not be reflected
+---in the editor immediately.
+---
+---Defaults to 2000.
+---@type number
 config.max_project_files = 2000
+
+---Enables/disables all transitions.
+---
+---Defaults to true.
+---@type boolean
 config.transitions = true
+
+---Enable/disable individual transitions.
+---These values are overriden by `config.transitions`.
 config.disabled_transitions = {
+  ---Enable/disables scrolling transitions.
```

Comment:
```suggestion
  ---Enables/disables scrolling transitions.
```
I guess it's more accurate to just say, `Disables ... transitions`.